examples/vision/oxford_pets_image_segmentation.py

```diff
@@ -1,113 +1,218 @@
 """
-Title: Image segmentation with a U-Net-like architecture
-Author: [fchollet](https://twitter.com/fchollet)
+Title: [KerasCV] Image segmentation with a U-Net-like architecture
+Author: [fchollet](https://twitter.com/fchollet), updated by [Aritra Roy
+Gosthipaty](https://twitter.com/ariG23498) and [Margaret
+Maynard-Reid](https://twitter.com/margaretmz)
+Reviewer: [Martin Gorner](https://twitter.com/martin_gorner)
 Date created: 2019/03/20
-Last modified: 2020/04/20
+Last modified: 2023/06/19
 Description: Image segmentation model trained from scratch on the Oxford Pets dataset.
 Accelerator: GPU
 """
+
 """
-## Download the data
+This tutorial uses a U-Net like architecture for image segmentation. Data processing and
+augmentations are implemented with [KerasCV](https://keras.io/keras_cv/).
+
+U-Net was introduced in the paper, [U-Net: Convolutional Networks for Biomedical Image
+Segmentation](https://arxiv.org/abs/1505.04597/). Although U-Net is a model for image
+segmentation, it's also used in generative models such as Pix2Pix and diffusion models.
+So it's important to have a solid understanding of its architecture.
 """
 
-"""shell
-curl -O https://thor.robots.ox.ac.uk/~vgg/data/pets/images.tar.gz
-curl -O https://thor.robots.ox.ac.uk/~vgg/data/pets/annotations.tar.gz
-tar -xf images.tar.gz
-tar -xf annotations.tar.gz
 """
+## Setup and Imports
 
+First let's set up install and imports of the dependencies.
 """
-## Prepare paths of input images and target segmentation masks
+
+import keras
+import keras_cv
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+import random
+import numpy as np
+from matplotlib import pyplot as plt
+
 """
+## Configuration
 
-import os
+Please feel free to tweak the configurations yourself and note how the model training
+changes. This is an excellent exercise to get more understanding of the training
+pipeline.
+"""
 
-input_dir = "images/"
-target_dir = "annotations/trimaps/"
-img_size = (160, 160)
+# Image Config
+height = 160
+width = 160
 num_classes = 3
-batch_size = 32
 
-input_img_paths = sorted(
-    [
-        os.path.join(input_dir, fname)
-        for fname in os.listdir(input_dir)
-        if fname.endswith(".jpg")
-    ]
-)
-target_img_paths = sorted(
-    [
-        os.path.join(target_dir, fname)
-        for fname in os.listdir(target_dir)
-        if fname.endswith(".png") and not fname.startswith(".")
-    ]
-)
+# Augmentation Config
+rotation_factor = (-0.2, 0.2)
 
-print("Number of samples:", len(input_img_paths))
+# Training Config
+batch_size = 128
+epochs = 50
+initial_learning_rate = 1e-4
+max_learning_rate = 5e-4
+warmup_epoch_percentage = 0.15
+AUTOTUNE = tf.data.AUTOTUNE
 
-for input_path, target_path in zip(input_img_paths[:10], target_img_paths[:10]):
-    print(input_path, "|", target_path)
+"""
+## Utility Functions
 
+The `unpackage_inputs` is a utility function that is later used to reformat the input
```

Comment:
This needs a bit more explaining:
- about the KerasCV dictionary format understood by preprocessing layers
- about the fact that the model will not understand that format and will expect a more traditional (image, target) pair
See where this explanation fits best in the text. You can move the "unpackage_inpuuts" function definition there.